### DISCUSSION TITLE
Smart contract definition

### DISCUSSION THREAD

**Discussion by AureliaDolo** [massalabs/massa#1944] [Documentation]
I see three things that we call "smart contract" and that are different, maybe we can find catchy name for all of them 

- executing an operation that contains plain byte code (for example that byte code could just make a coin transfer to another address) that won't be saved in the ledger
- the byte code that is stored on an address that is derived from a private key that was stored through a write call on the user's ledger space
- the byte code that is stored on an address that was created through a create smart contract call, that is not derived from a private key and that does not have an owner

**Comment by damip** [comment]
Bytecode : bytes that can be executed by the VM

Bytecode can be stored in : 
* ExecuteSC operations: when the VM executes it, it runs it within the context of the sender address
* The ledger code space of a "normal" address (with private key) or "smart contract" address (without private key)  A: can be imported by any running bytecode for local execution. Its public exported functions can be called by any bytecode, which will execute them within the context of A.

**Comment by AureliaDolo** [comment]
So what actually is a smart contract ?

**Comment by damip** [comment]
If you really want to make it formal, let's say that we never say "smart contract" but only "smart contract address" or "normal address", and never use it in any other context
